### PR TITLE
#167165081 Users should be able to highlight and comment on any text in an article

### PR DESCRIPTION
**Pull request by tolulope-od**
### What does this PR do?
Implements highlighting of text in an article and comment on it
Adds Swagger documentation
Adds `highlightedText` with `TEXT` data-type  and `containsHighlightedText` with `BOOLEAN` data-type columns to the Comments table
Refactors code to fetch single article with comments
Adds more test cases

#### Description of Task to be completed?
A user should be able to highlight a text in an article and comment on through the API route
`POST -> /api/v1/articles/:slug/comments`

*Flowchart for the comment creation process*
Like a Comment API Flowchart (5)

#### How should this be manually tested?

After cloning this repo, `cd` into it and do the following:

```bash
# fetch this branch
$ git fetch origin  ft-highlight-and-comment-on-text-167165081

# install dependencies
$ npm install

# run unit tests
$ npm test

# start the development server
$ npm run server
```
Using postman, log in or register an account and hit the post route to create an article (user must be verified via the verification route in #28 
After getting the token, add it to the header and make a request to the route stated above

#### Any background context you want to provide?
N/a

#### What are the relevant pivotal tracker stories?
[#167165081](https://www.pivotaltracker.com/story/show/167165081)

#### Screenshots (if appropriate)

*Successfully Create a comment with highlighted text*
Screenshot 2019-08-13 at 21 14 30
Screenshot 2019-08-14 at 11 05 51 AM

### CODE REVIEW

**Review by mekzy-o — APPROVED** [review]
Good Job guys

File: docs/articles/postComment.json [hunk 1]
```diff
@@ -19,6 +19,10 @@
             "body": {
               "type": "string",
               "example": "just some comment"
+            },
+            "highlightedText": {
```
Comment (ocranbillions) [inline docs/articles/postComment.json:23]: Nice job guys. I like the thought process for implementing this. 
So I noticed you forgot to include the `highlightedText` field in the attributes section where we included comments on fetchArticles. Without this, we can't access this field on comments that have highlight text.

**Review by ocranbillions — APPROVED** [review]
LGTM. Kindly make the changes and push again or let me know your thoughts.

File: controllers/ArticleController.js [hunk 2]
```diff
@@ -99,7 +99,7 @@ export default class ArticleController {
           { // Article Comments
             model: Comment,
             as: 'articlecomment',
-            attributes: ['body', 'likes', 'createdAt', 'updatedAt'],
+            attributes: ['body', 'likes', 'highlightedText', 'containsHighlightedText', 'createdAt', 'updatedAt'],
             include: [
```
Comment (ocranbillions) [inline controllers/ArticleController.js:103]: So sorry guys. I see you've implemented it here in the getArticleBySlug method. My bad

File: docs/articles/postComment.json (hunk 1 above)
Comment (PelumiAlesh) [reply docs/articles/postComment.json:23]: We are only fetching the comments when we fetch single article. We don't need to get comments when fetching all articles.

Comment (ocranbillions) [reply docs/articles/postComment.json:23]: Cool. That totally makes sense.

File: .travis.yml [hunk 3]
```diff
@@ -1,5 +1,6 @@
+language: node_js
 node_js:
-  - '10'
+  - 'stable'
```
Comment (OdunayoOkebunmi) [inline .travis.yml:3]: I see you have fixed the Travis issue. Good job

**Review by OdunayoOkebunmi — APPROVED** [review]

**Pull request merged**